org.jacoco.core/src/org/jacoco/core/internal/flow/Instruction.java

```diff
@@ -65,20 +69,35 @@ public void addBranch() {
 	 * @see #addBranch()
 	 * @param predecessor
 	 *            predecessor instruction
+	 * @param branch
+	 *            branch number in predecessor that should be marked as covered
+	 *            when this instruction marked as covered
 	 */
-	public void setPredecessor(final Instruction predecessor) {
+	public void setPredecessor(final Instruction predecessor,
```

Comment:
Unrelated to this change but while trying to understand my own code after some time I wondered, why a instruction can't have multiple predecessors: While this is of course possible in byte code JaCoCo adds probes in a way that each instruction has at most one predecessor. Otherwise probes are inserted. Maybe we can improve documentation here:

> Sets the given instruction as a direct predecessor to this instruction. Every instruction can have at most one direct predecessor otherwise probes are inserted in-between. This will add an branch to the predecessor.

What about adding a assertion to enforce this invariant?

		if (this.predecessor != null) {
			throw new AssertionError();
		}